src/components/ClaimENSPage/ClaimENSPage.container.ts

```diff
@@ -1,20 +1,33 @@
 import { connect } from 'react-redux'
-import { push, goBack } from 'connected-react-router'
+import { push, goBack, getLocation, replace } from 'connected-react-router'
 import { getData as getWallet, getMana } from 'decentraland-dapps/dist/modules/wallet/selectors'
 import { openModal } from 'modules/modal/actions'
 import { RootState } from 'modules/common/types'
+import { hasHistory } from 'modules/location/selectors'
+import { DeployToWorldLocationStateProps, FromParam } from 'modules/location/types'
 import { MapDispatchProps, MapDispatch, MapStateProps } from './ClaimENSPage.types'
 import ClaimENSPage from './ClaimENSPage'
 
 const mapState = (state: RootState): MapStateProps => {
+  const isFromDeployToWorld = (getLocation(state).state as DeployToWorldLocationStateProps)?.fromParam === FromParam.DEPLOY_TO_WORLD
```

Comment:
do we want to maintain this redirection if the user refreshes the page? maybe add it as query param?